### PR TITLE
https://github.com/bbfsdev/bbfs/issues/198

### PR DESCRIPTION
**Pull request by AlexeyNemytov**
do not write data to remoteif not changed

### CODE REVIEW

File: lib/content_server/remote_content.rb [hunk 1]
```diff
@@ -37,10 +38,15 @@ def receive_content(message)
...
         write_to = File.join(@content_server_content_data_path,
                              @last_save_timestamp.to_s + '.cd')
-        FileUtils.makedirs(@content_server_content_data_path) unless \
+        same_message = message.to_s.eql?(@last_message.to_s)  #check if same message received
+        if(!same_message)
+          FileUtils.makedirs(@content_server_content_data_path) unless \
               File.directory?(@content_server_content_data_path)
-        count = File.open(write_to, 'wb') { |f| f.write(message.to_s) }
-        Log.debug1("Written content data to file:#{write_to}.")
+          count = File.open(write_to, 'wb') { |f| f.write(message.to_s) }
```
Comment (genadyp) [inline lib/content_server/remote_content.rb]: Do we need count variable here? it looks unused.

File: lib/content_server/remote_content.rb [hunk 2]
```diff
@@ -37,10 +38,15 @@ def receive_content(message)
       	@last_save_timestamp = Time.now.to_i
         write_to = File.join(@content_server_content_data_path,
                              @last_save_timestamp.to_s + '.cd')
-        FileUtils.makedirs(@content_server_content_data_path) unless \
+        same_message = message.to_s.eql?(@last_message.to_s)  #check if same message received
```
Comment (genadyp) [inline lib/content_server/remote_content.rb]: Is last_message is a ContentData object? If so, suggest to use [ContentData::==](https://github.com/bbfsdev/bbfs/blob/master/lib/content_data/content_data.rb#L204).

same_message is a confusing name, cause there are variables that contains  "message" string and that are ContentData object.
Consider to rename it, for example, to is_same_message, cause this name intends to boolean.
or move check to the "if", one line beneath.

**Comment by genadyp** [comment]
@AlexeyNemytov 
Alexey, add, please short methods description for the methods that you used.
Suggest [YARD](http://rubydoc.info/docs/yard/file/docs/GettingStarted.md) for this purpose.

It's enough to do something like:

``` ruby
# class description...
class Request
# code...
end

class MyWebServer
  # Handles a request
  # @param request [Request] the request object
  # @return [String] the resulting webpage
  def get(request) "hello" end

  # (see #get)
  def post(request) "hello" end
end
```

**Comment by AlexeyNemytov** [comment]
nice one

File: lib/content_server/remote_content.rb [hunk 3]
```diff
@@ -17,6 +17,7 @@ def initialize(dynamic_content_data, host, port, local_backup_folder)
       @remote_tcp = Networking::TCPClient.new(host, port, method(:receive_content))
       @last_fetch_timestamp = nil
       @last_save_timestamp = nil
+      @last_message = nil
```
Comment (yarondbb) [inline lib/content_server/remote_content.rb]: concider rename it to last_content_data for better readility

Comment (yarondbb) [inline lib/content_server/remote_content.rb]: saving the entire last content data object is memory consuming (might be few mega bytes in memory for big objects).
I recommend using other "smart" solution to check if last content data has changed.

File: lib/content_server/remote_content.rb (hunk 2 above)
Comment (AlexeyNemytov) [inline lib/content_server/remote_content.rb]: changed usage to ContentData::==.

File: lib/content_server/remote_content.rb (hunk 1 above)
Comment (AlexeyNemytov) [inline lib/content_server/remote_content.rb]: removed

File: lib/content_server/remote_content.rb (hunk 3 above)
Comment (AlexeyNemytov) [inline lib/content_server/remote_content.rb]: @yarondbb 
already commited a new changes , just saw your comments
please check a new code
will think about 'smart' solution :)

Comment (yarondbb) [inline lib/content_server/remote_content.rb]: Where is th enew code? it should be part of this pull request. Did you push it in same branch?

File: lib/content_server/remote_content.rb [hunk 4]
```diff
@@ -37,15 +53,22 @@ def receive_content(message)
       	@last_save_timestamp = Time.now.to_i
         write_to = File.join(@content_server_content_data_path,
                              @last_save_timestamp.to_s + '.cd')
-        FileUtils.makedirs(@content_server_content_data_path) unless \
+
+        is_same_content = (message==@last_message)  #check if same ContentData received
+        if(!is_same_content)
```
Comment (yarondbb) [inline lib/content_server/remote_content.rb]: could combine lines 57 and 58 to be:
unless message == @last_message
..

File: lib/content_server/remote_content.rb [hunk 5]
```diff
@@ -37,15 +53,22 @@ def receive_content(message)
...
+          FileUtils.makedirs(@content_server_content_data_path) unless \
               File.directory?(@content_server_content_data_path)
-        count = File.open(write_to, 'wb') { |f| f.write(message.to_s) }
-        Log.debug1("Written content data to file:#{write_to}.")
+          File.open(write_to, 'wb') { |f| f.write(message.to_s) }
```
Comment (yarondbb) [inline lib/content_server/remote_content.rb:60]: I suggest using to_file method of ContentData (message is actually an instance ContentData)

**Comment by AlexeyNemytov** [comment]
@yarondbb 
@genadyp 
Please check new commits

**Comment by yarondbb** [comment]
Greate documentation Alexey!!!
I have merged this code manually (due to many conflicts)

**Pull request closed without merging**